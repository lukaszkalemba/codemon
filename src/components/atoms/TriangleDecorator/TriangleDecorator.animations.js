const getTriangle = isMobile => {
  return {
    variants: {
      initial: {
        opacity: 0,
        x: isMobile ? -25 : 50,
        y: isMobile ? -25 : 0,
      },
      animate: {
        opacity: 1,
        x: 0,
        y: 0,
        transition: {
          duration: 0.75,
        },
      },
    },
    initial: 'initial',
    animate: 'animate',
  };
};

export default {
  getTriangle,
};
