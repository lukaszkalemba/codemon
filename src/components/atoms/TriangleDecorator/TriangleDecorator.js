import React from 'react';
import PropTypes from 'prop-types';
import S from './TriangleDecorator.styles';

const TriangleDecorator = ({ animation }) => {
  return <S.TriangleDecorator {...animation} />;
};

TriangleDecorator.defaultProps = {
  animation: null,
};

TriangleDecorator.propTypes = {
  animation: PropTypes.shape({
    variants: PropTypes.shape({
      animate: PropTypes.shape({}),
      initial: PropTypes.shape({}),
    }),
  }),
};

export default TriangleDecorator;
