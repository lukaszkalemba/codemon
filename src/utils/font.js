export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
