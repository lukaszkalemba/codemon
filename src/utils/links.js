export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
