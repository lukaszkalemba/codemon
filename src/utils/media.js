export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
