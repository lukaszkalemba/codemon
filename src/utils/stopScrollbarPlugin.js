import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
