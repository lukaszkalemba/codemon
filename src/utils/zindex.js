export default {
  negative5: '-5000',
  negative4: '-4000',
  negative3: '-3000',
  negative2: '-2000',
  negative1: '-1000',
  zero: '0',
  positive1: '1000',
  positive2: '2000',
  positive3: '3000',
  positive4: '4000',
  positive5: '5000',
};
