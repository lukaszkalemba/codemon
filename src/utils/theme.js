import color from 'utils/color';
import font from 'utils/font';
import media from 'utils/media';
import zindex from 'utils/zindex';

export default {
  color,
  font,
  media,
  zindex,
};
