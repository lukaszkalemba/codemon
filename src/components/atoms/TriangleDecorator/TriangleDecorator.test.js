import React from 'react';
import { render } from 'utils/test-utils';
import TriangleDecorator from './TriangleDecorator';

describe('<TriangleDecorator />', () => {
  it('matches snapshot', () => {
    const { container } = render(<TriangleDecorator />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
