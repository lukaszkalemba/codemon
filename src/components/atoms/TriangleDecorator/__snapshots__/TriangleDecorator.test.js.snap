// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TriangleDecorator /> matches snapshot 1`] = `
<div
  class="sc-bdnylx kOpDmn"
  style="opacity: 0; transform: translateX(-25px) translateY(-25px) translateZ(0);"
/>
`;
